src/dataregistry/registrar/dataset.py

```diff
@@ -606,10 +708,10 @@ def _find_previous(self, name, version_string, version_suffix, owner, owner_type
             result = conn.execute(stmt)
             conn.commit()
 
-        # Pull out the single result
+        # There should only be a single result that is not overwritten
```

Comment:
It's true that there *should* only be one, but it seems possible - albeit very unlikely - that if some earlier attempt to overwrite didn't complete properly or if the code was buggy that there could be more.  To be safe, I would add a clause to the query to order by `replace_iteration`, descending, and let the `replace_iteration`return value be a list.   Then `replace` will be able to find the correct next value for `replace_iteration` easily and, if there was some earlier issue with the value of `is_overwritten`, it can be silently fixed (or, perhaps better,  issue a warning message).   With the current code the replace will fail and the only way to fix it will be for the user to notify one of us to fix it manually.

Reply:
`find_previous` now returns a dict, with the list entries ordered by `replace_iteration`.

Also, only `valid==1` datasets can now be replaced, i.e., valid but non-deleted, non-archived datasets.

Added a test to make sure deleted datasets cannot be replaced.

Added a test to make sure you cant replace is_overwritable=False